test/unit/io/sequence/sequence_file_out_test.cpp

```diff
@@ -405,6 +426,34 @@ TEST(rows, assign_sequence_file_pipes)
     EXPECT_EQ(fout.get_stream().str(), input);
 }
 
+TEST(rows, convert_fastq_to_fasta)
+{
+    std::string const fastq_in
+    {
+    "@ID1\n"
+    "ACGTT\n"
+    "+\n"
+    "!##$%\n"
+    "@ID2\n"
+    "TATTA\n"
+    "+\n"
+    ",BDEB\n"
+    };
+
+    std::string const fasta_out
+    {
+        "> ID1\n"
+        "ACGTT\n"
+        "> ID2\n"
+        "TATTA\n"
+    };
+
+    auto fout = sequence_file_in{std::istringstream{fastq_in}, sequence_file_format_fastq{}} |
+                sequence_file_out{std::ostringstream{}, sequence_file_format_fasta{}};
+    fout.get_stream().flush();
+    EXPECT_EQ(fout.get_stream().str(), fasta_out);
+}
+
```

Comment:
I think this is covered by the previous test or not? Format specific things should not be tested here, but in the format tests...

Reply:
Here I disagree, as it tests the sequence_file_out ability of converting file formats rather than fastq specific things. The fastq format test does not know about sequence_file_out. 
It is not covered in the previous test, because there the format and fields are the same; it rather extends it.